### DISCUSSION TITLE
Time based events

### DISCUSSION THREAD

**Discussion by GeekyG4mer** [rafaelvcaetano/melonDS-android#329] [Support]
Are time based events like growing berries in pokemon functional? If yes is it enabled by default or I would have to tweak some settings?

**Comment by rafaelvcaetano** [comment]
They should work out of the box. However, if you enable the "Randomize MAC address" option in either the "Firmware configuraron" or in the "Custom BIOS and firmware" screens, they may stop working.

**Reply by GeekyG4mer** [reply]
I accidentally enabled randomize mac address for a sec, but I disabled it immediately. Will my berries grow ever again?

**Reply by rafaelvcaetano** [reply]
I think it should not be a problem. Basically, enabling that option makes the games think they are running on different systems each time. So, after a couple of hours, your berries should start growing.

**Reply by GeekyG4mer** [reply]
I waited three days, but none of my berries grew and I noticed daily events aren't happening, meaning that the game think a day has not passed, or worse no time has passed. The day-night cycle is fine tho. Randomise mac address is currently disabled. Maybe it is a bug. Please look into it.

**Reply by rafaelvcaetano** [reply]
Alright. Thanks for keeping me updated. I'll look into this.

**Reply by GeekyG4mer** [reply]
Any leads aur updates worth sharing. The time has not passed even now.

**Reply by rafaelvcaetano** [reply]
Not yet. I've been looking into it but still haven't been able to replicate the problem. I still have a couple of tests I want to make, so, I'll let you know once I find something suspicious.

**Reply by GeekyG4mer** [reply]
Hey, any updates?

**Reply by rafaelvcaetano** [reply]
Unfortunately, not. Whatever I try, berries and other time based events always work as expected. I know you have already confirmed this to me, but could you send me screenshots from both your "Internal firmware settings" and "Custom BIOS and firmware"?

**Reply by ashgx** [reply]
> I waited three days, but none of my berries grew and I noticed daily events aren't happening, meaning that the game think a day has not passed, or worse no time has passed. The day-night cycle is fine tho. Randomise mac address is currently disabled. Maybe it is a bug. Please look into it.

I use  melonds on PC. When MAC address randomization is prohibited, I have to re open the game and save it; The next day, the fruit in the Pokemon heart gold will grow normally. I don't know if it's the same with mobile phones? You can try it this way. I suggest that if you don't use pokemon exchange, you'd better prohibit random MAC address.

**Reply by GeekyG4mer** [reply]
Update : I have found a fix online for daily events. Turn the time to 23:59 and let it roll over to the next day and save. This fixed daily events. 
             I will have to wait to recommend this fix as :
    1. The game may break again as I messed with the system clock
    2. I still have to see if the berries start growing.
    3. There is no way to change time from the emulator itself so you have to change your system time. I don't know if it causes any problems but it may. [ Please add a feature to change the emulator's time ].
    So I will give you a update about the berries tomorrow.
Currently running on beta 1.6 on android 11
